Source/WebCore/platform/mock/MockRealtimeVideoSource.h

```diff
@@ -60,7 +60,7 @@ class MockRealtimeVideoSource : public RealtimeVideoCaptureSource, private Orien
 protected:
     MockRealtimeVideoSource(String&& deviceID, AtomString&& name, MediaDeviceHashSalts&&, std::optional<PageIdentifier>);
 
-    virtual void updateSampleBuffer() = 0;
+    virtual void updateSampleBuffer() { };
```

Comment:
How does it work if updateSampleBuffer() does nothing?
Ditto for audio.

Reply:
This won't push buffers, but this is the same status than before 213468@main

This is a build fix to be able to build MediaStream without GStreamer support, but these Realtime mock sources won't work at runtime.

Reply:
I don't see much the point then. You make the mock sources unusable only for build fixes sake?

If you're not using GStreamer or Cocoa, you should add some new mock sources for whatever your platform uses.

Reply:
I would vote for 3. wdyt @youennf and @eric-carlson ?